src/mtcp/mtcp_restart.c

```diff
@@ -1112,38 +1112,7 @@ read_one_memory_area(int fd, VA endOfStack)
         mmap_fixed_noreplace(area.addr, area.size, area.prot | PROT_WRITE,
                             area.flags, imagefd, area.offset);
 
-      if (mmappedat == MAP_FAILED) {
```

Comment:
I remember you saying that we hit an error in mtcp_restart.c, and we print the error, but try to continue anyway.
But I'm not sure that this code is doing what you said.  For example, I'm reading that if this is a zero page, then it's not really an error, and we really should be continuing.
Please remove this commit from the current PR.  If there's some worthwhile cleanup to do, please submit this single commit as a new PR, but not part of the current PR.